module/admin/helper.js

```diff
@@ -135,4 +145,413 @@ module.exports = class adminHelper {
+
+          return resolve({
+            success: true,
+            message: messageConstants.apiResponses.SOLUTION_RESOURCE_DELETED,
+            result: {
+              solutionDeletedCount,
+              surveyCount,
+              surveySubmissionCount,
+              observationCount,
+              observationSubmissionCount,
+              pullSolutionFromProgramComponent,
+            },
+          });
+        } else {
+          return resolve({
+            success: false,
+            message: messageConstants.apiResponses.INVALID_RESOURCE_TYPE,
+          })
+        }
+      } catch (error) {
+        return resolve({
+          success: false,
+          message: error.message,
+          status:error.status,
+          data: false,
+        });
+      }
+    });
+  }
+
+  /**
+ * Deletes associated survey and observation resources based on solution details and tenant ID.
+ * @method
+ * @name deleteAssociatedResources
+ * @param {Array<{ _id: string, type: string }>} solutionDetails - List of solution objects with `_id` and `type` (e.g., SURVEY or OBSERVATION).
+ * @param {string} tenantId - Tenant identifier.
+  */
+  static deleteAssociatedResources(solutionDetails, tenantId) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let surveyCount = 0;
+        let surveySubmissionCount = 0;
+        let observationCount = 0;
+        let observationSubmissionCount = 0;
+
+        const surveyIds = [];
+        const observationIds = [];
+        // Categorize solution types into survey/observation
+        for (const solutionType of solutionDetails) {
+          if (solutionType.type === messageConstants.common.SURVEY) {
+            surveyIds.push(solutionType._id);
+          } else if (solutionType.type === messageConstants.common.OBSERVATION) {
+            observationIds.push(solutionType._id);
+          }
+        }
+        // Delete survey documents and submissions
+        if (surveyIds.length) {
+          const surveyFilter = { solutionId: { $in: surveyIds }, tenantId };
+          const surveyDetails = await surveyQueries.surveyDocuments(surveyFilter, ['_id']);
+
+          if (surveyDetails?.length) {
+            surveyCount = surveyDetails.length;
+            await surveyQueries.delete(surveyFilter);
+          }
+
+          const surveySubmissionDetails = await surveySubmissionQueries.surveySubmissionDocuments(surveyFilter, [
+            '_id',
+          ]);
+
+          if (surveySubmissionDetails?.length) {
+            surveySubmissionCount = surveySubmissionDetails.length;
+            await surveySubmissionQueries.delete(surveyFilter);
+          }
+        }
+
+        // Delete observation documents and submissions
+        if (observationIds.length) {
+          const observationFilter = { solutionId: { $in: observationIds }, tenantId };
+          const observationDetails = await observationQueries.observationDocuments(observationFilter, ['_id']);
```

Comment:
@MallanagoudaB why we are fetching the data if we intent to delete them? is there any reason for this